### PR TITLE
Upgrade to terraform 1.2.9

### PR DESCRIPTION
**Pull request by omissis**
- chore: update all modules and examples to support the latest terraform and providers
- chore: make examples running directly using terraform
- fix: example vpn addresses

### CODE REVIEW

File: example/eks/main.tf [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+  cluster_version = "1.23"
 
-  network         = "vpc-id0"
-  subnetworks = [
-    "subnet-id1",
-    "subnet-id2",
-    "subnet-id3",
-  ]
+  network     = data.terraform_remote_state.vpc_and_vpn.outputs.vpc_id
+  subnetworks = data.terraform_remote_state.vpc_and_vpn.outputs.public_subnets
```
Comment (lnovara) [inline example/eks/main.tf:15]:
```suggestion
  subnetworks = data.terraform_remote_state.vpc_and_vpn.outputs.private_subnets
```

**Review by lnovara — CHANGES_REQUESTED** [review]
Some minor comments.

LGTM :shipit:

File: example/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# TASK: fill in main.auto.tfvars with your data
+terraform init
+terraform apply
+
+# Once all the above is done you can dump the kube config to a file of your choice
+terraform output -raw kubeconfig > /var/tmp/.kubeconfig
+
+# Last but not least, you can verify your cluster is up and running
+KUBECONFIG=/var/tmp/.kubeconfig kubectl get nodes
+```
```
Comment (lnovara) [inline example/README.md:37]: I would add a `terraform destroy` at the end in both examples.

File: example/vpc-and-vpn/main.auto.tfvars.dist [hunk 3]
```diff
@@ -0,0 +1 @@
+vpn_ssh_users = []
```
[the remark this answers is not in the document]
Comment (omissis) [reply example/vpc-and-vpn/main.auto.tfvars.dist:2]: I added a comment above it so to leave the file without errors at runtime should it be forgotten during renaming

**Review by lnovara — APPROVED** [review]
Just add the `terraform destroy` at the end of the example README and we can merge

**Pull request merged**